crates/kornia-pnp/src/ippe.rs

```diff
@@ -0,0 +1,160 @@
+//! Infinitesimal Plane-based Pose Estimation (IPPE)
+//!
+//! This module provides routines to estimate two candidate poses for a planar
+//! object from 2D to 3D correspondences, following the IPPE method.
+//!
+//! References:
+//! - T. Collins and A. Bartoli, "Infinitesimal Plane-based Pose Estimation"
+//! - OpenCV implementation reference: modules/calib3d/src/ippe.hpp/.cpp
+//!
+//! This file contains original Rust code inspired by the algorithms described
+//! in the references above. The OpenCV implementation is BSD-3-Clause; see the
+//! OpenCV project license for details.
+
+use crate::pnp::{PnPError, PnPResult};
+use glam::{Mat3A, Vec3, Vec3A};
+use kornia_lie::so3::SO3;
+
+/// Two IPPE pose solutions sorted by reprojection error (best first).
+///
+/// Note: The current implementation returns the same solution twice `(best, second)`
+/// as a temporary placeholder until the IPPE-specific second solution is implemented.
+pub type IPPEResult = (PnPResult, PnPResult);
+
+/// Marker type for the IPPE solver.
+pub struct IPPE;
+
+impl IPPE {
+    /// Estimate two candidate poses for a square planar object from its four image corners.
+    ///
+    /// The four object points are implicitly defined as a square of side length `square_length`
+    /// on the plane z=0 with the following order (object coordinates):
+    ///  - p0 = [-L/2,  L/2, 0]
+    ///  - p1 = [ L/2,  L/2, 0]
+    ///  - p2 = [ L/2, -L/2, 0]
+    ///  - p3 = [-L/2, -L/2, 0]
+    ///
+    /// The `image_points_norm` are the corresponding undistorted, normalized image coordinates
+    /// (i.e. pixel coordinates premultiplied by K^{-1}).
+    pub fn solve_square(
+        image_points_norm: &[[f32; 2]; 4],
```

Comment:
for api consistency with the pnp module, this should accept unnormalized image coordinates and accept the camera intrinsics. Check the epnp `solve_epnp` function. Change name to `points_image` too